packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m

```diff
@@ -16,6 +16,19 @@ - (instancetype)initWithChannel:(FlutterMethodChannel *)channel {
   return self;
 }
 
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (![keyPath isEqualToString:@"URL"]) {
+    return;
+  }
+  NSURL *newURL = [change objectForKey:NSKeyValueChangeNewKey];
+  if (newURL) {
```

Comment:
There should be a unit test verifying that the method channel is not invoked when the `newUrl` parameter is `nil`.

Reply:
Added